lib/hint.rb

```diff
@@ -16,34 +16,43 @@
 # you may find current contact information at www.suse.com
 
 class Hint
-  def self.get_started
-    output "You can get started by inspecting a system. Run:\n#{$0} inspect HOSTNAME"
+  def self.print(method, options = {})
+    if Machinery::Config.new.hints
+      Machinery::Ui.puts to_string(method, options)
+    end
+  end
+
+  def self.to_string(method, options = {})
+    Machinery::Config.new.hints ? "\nHint: #{send(method, options)}\n" : ""
+  end
+
+  private
+
+  def self.get_started(options)
+    "You can get started by inspecting a system. Run:\n#{$0} inspect HOSTNAME"
+  end
+
+  def self.upgrade_format_force(options)
+    "To force an upgrade of system descriptions run:\n" \
+    "#{$0} upgrade-format --force #{options[:name]}"
   end
 
   def self.show_data(options)
-    output "To show the data of the system you just inspected run:\n#{$0} show #{options[:name]}"
+    "To show the data of the system you just inspected run:\n#{$0} show #{options[:name]}"
   end
 
   def self.show_analyze_data(options)
-    output "To show the config file diffs you just created run:\n" \
+    "To show the config file diffs you just created run:\n" \
       "#{$0} show --scope config-files --show-diffs #{options[:name]}"
   end
 
   def self.do_complete_inspection(options)
-    output "To do a full inspection containing all scopes and to extract files run:\n" \
+    "To do a full inspection containing all scopes and to extract files run:\n" \
      "#{$0} inspect #{options[:host]} --name #{options[:name]} --extract-files"
   end
 
-  def self.upgrade_system_description
-    output "To upgrade all system descriptions run:\n" \
+  def self.upgrade_system_description(options)
```

Comment:
Unused method argument - `options`. If it's necessary, use `_` or `_options` as an argument name to indicate that it won't be used. You can also write as `upgrade_system_description(*)` if you want the method to accept any arguments but don't care about them.